dask/dataframe/multi.py

```diff
@@ -902,6 +902,19 @@ def merge_asof(
             " 'forward', or 'nearest'"
         )
 
+    if left is None or right is None:
+        raise ValueError("Cannot merge_asof on None")
+
+    if (left_on is not None or right_on is not None) and on is not None:
+        raise ValueError(
+            "Can only pass argument 'on' OR 'left_on' and 'right_on', not a combination of both."
+        )
```

Comment:
Should the `on` logic include `left_index` and `right_index` checks? I think this what happens in pandas [here](https://github.com/pandas-dev/pandas/blob/d1accd032b648c9affd6dce1f81feb9c99422483/pandas/core/reshape/merge.py#L1561-L1566).

Reply:
I don't quite get the logic that pandas includes there. I think pandas supports `*_on` options to be lists in case one index or the other is a `MultiIndex`, which Dask doesn't support. In other words, I'm not sure a change is needed here? Maybe I'm just misunderstanding what `merge_asof()` does in pandas.

Reply:
Oh you are completely right! I misunderstood those lines in pandas. My point was that the combination of `left_on` and `right_index` is possible (e.g. [this test](https://github.com/dask/dask/blob/e49880b67990c6f74f3ec034cff79623bb5ce21c/dask/dataframe/tests/test_multi.py#L331-L358)), so I was wondering if the logic on lines 934-937 should include that option.